static/js/peer_data.js

```diff
@@ -94,38 +94,61 @@ export function get_subscribers(stream_id) {
 export function set_subscribers(stream_id, user_ids) {
     const subscribers = new LazySet(user_ids || []);
     stream_subscribers.set(stream_id, subscribers);
+
+    return subscribers;
 }
 
 export function add_subscriber(stream_id, user_id) {
     const subscribers = stream_subscribers.get(stream_id);
     if (typeof subscribers === "undefined") {
         blueslip.warn("We got an add_subscriber call for an untracked stream: " + stream_id);
-        return false;
+        return;
     }
     const person = people.get_by_user_id(user_id);
     if (person === undefined) {
         blueslip.error("We tried to add invalid subscriber: " + user_id);
-        return false;
+        return;
     }
     subscribers.add(user_id);
+}
+
+export function bulk_add_subscribers({stream_ids, user_ids}) {
+    // We trust the server data here.  Worst case scenario is
+    // that we just track a stream that will never be referenced.
+    stream_ids.forEach((stream_id) => {
+        const subscribers = stream_subscribers.get(stream_id) || set_subscribers(stream_id);
 
-    return true;
+        user_ids.forEach((user_id) => {
+            subscribers.add(user_id);
+        });
+    });
 }
 
 export function remove_subscriber(stream_id, user_id) {
     const subscribers = stream_subscribers.get(stream_id);
     if (typeof subscribers === "undefined") {
         blueslip.warn("We got a remove_subscriber call for an untracked stream " + stream_id);
-        return false;
+        return;
     }
     if (!subscribers.has(user_id)) {
         blueslip.warn("We tried to remove invalid subscriber: " + user_id);
-        return false;
+        return;
     }
 
     subscribers.delete(user_id);
+}
 
-    return true;
+export function bulk_remove_subscribers({stream_ids, user_ids}) {
+    // We trust the server data here.  If we don't have
+    // a stream_id/user_id tuple we just fail silently.
+    stream_ids.forEach((stream_id) => {
```

Comment:
I feel like we've generally been aiming to standardize on `for ... of ` loops; is there a reason to prefer `forEach` in these modules?

@andersk FYI.

Reply:
I have no preference here, and I see that @andersk added lint rules to enforce the for/of standard, so I'll address that as I re-push.